nixos/modules/programs/fish.nix

```
@@ -53,7 +55,23 @@ in
         type = lib.types.bool;
       };
 
-      package = lib.mkPackageOption pkgs "fish" { };
+      package = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs.fish;
+        defaultText = lib.literalExpression "pkgs.fish";
+        apply =
+          fish:
+          if fish ? unwrapped then
+            fish.override { pluginPkgs = cfg.plugins; }
```

```suggestion
            fish.override { pluginPkgs = cfg.plugins ++ lib.optionals (fish ? pluginPkgs) fish.pluginPkgs; }
```

You discarded the plugin that the user originally set in the package, see: https://github.com/NixOS/nixpkgs/blob/9f918d616c5321ad374ae6cb5ea89c9e04bf3e58/nixos/modules/programs/steam.nix#L58C46-L58C97